examples/demuxer_isom.exs

```diff
@@ -0,0 +1,74 @@
+Mix.install([
+  :membrane_aac_plugin,
+  :membrane_h264_ffmpeg_plugin,
+  :membrane_hackney_plugin,
+  {:membrane_mp4_plugin, path: __DIR__ |> Path.join("..") |> Path.expand()}
+])
+
+defmodule Example do
+  use Membrane.Pipeline
+
+  @input_file "https://raw.githubusercontent.com/membraneframework/static/gh-pages/samples/big-buck-bunny/bun33s.mp4"
+  @output_video "example.h264"
+  @output_audio "example.aac"
+
+  @impl true
+  def handle_init(_ctx, _opts) do
+    structure = [
+      child(:video_source, %Membrane.Hackney.Source{
+        location: @input_file,
+        hackney_opts: [follow_redirect: true]
+      })
+      |> child(:demuxer, Membrane.MP4.Demuxer.ISOM)
+      |> via_out(Pad.ref(:output, 1))
+      |> child(:depayloader_video, Membrane.MP4.Depayloader.H264)
+      |> child(:sink_video, %Membrane.File.Sink{location: @output_video}),
+      get_child(:demuxer)
+      |> via_out(Pad.ref(:output, 2))
+      |> child(:depayloader_audio, Membrane.MP4.Depayloader.AAC)
+      |> child(:audio_parser, %Membrane.AAC.Parser{
+        in_encapsulation: :none,
+        out_encapsulation: :ADTS
+      })
+      |> child(:sink_audio, %Membrane.File.Sink{location: @output_audio})
+    ]
+
+    {[spec: structure, playback: :playing], %{audio_processed?: false, video_processed?: false}}
+  end
+
+  # The rest of the module is used only for pipeline self-termination after processing finishes
+  @impl true
+  def handle_element_end_of_stream(:sink_video, _pad, _ctx, state) do
+    state = %{state | video_processed?: true}
+
+    actions =
+      if state.audio_processed? and state.video_processed?, do: [terminate: :shutdown], else: []
+
+    {actions, state}
+  end
+
+  @impl true
+  def handle_element_end_of_stream(:sink_audio, _pad, _ctx, state) do
+    state = %{state | audio_processed?: true}
+
+    actions =
+      if state.audio_processed? and state.video_processed?, do: [terminate: :shutdown], else: []
+
+    {actions, state}
+  end
+
+  @impl true
+  def handle_element_end_of_stream(_child, _pad, _ctx, state) do
+    {[], state}
+  end
```

Comment:
can we make it a little bit less verbose?

Reply:
done